ulid/__init__.py

```diff
@@ -71,7 +74,7 @@ def __init__(self, value: bytes | None = None) -> None:
 
     @classmethod
     @validate_type(datetime)
-    def from_datetime(cls, value: datetime) -> ULID:
+    def from_datetime(cls: type[U], value: datetime) -> U:
```

Comment:
I think this is the way to do such annotations: https://typing.readthedocs.io/en/latest/spec/generics.html#self